src/coreclr/jit/morph.cpp

```diff
@@ -6373,7 +6373,13 @@ GenTree* Compiler::fgMorphCall(GenTreeCall* call)
         if (call->gtCallType == CT_INDIRECT)
         {
             optCallCount++;
-            optIndirectCallCount++;
+            // Do not count the tailcall morphed from fgMorphCall ->
+            // fgMorphPotentialTailCall -> fgMorphCall as indirect call
+            // to prevent forcing EBP frame later.
```

Comment:
nit: maybe arch neutral terminology would be better here.
```suggestion
            // to avoid unnecessarily forcing a frame-pointer-based stack frame later.
```

Reply:
The name is not ideal, it matches `Compiler::rpMustCreateEBPFrame` though. Let's see what the JIT team tells me, maybe I will need to rewrite it in a different way anyway...